makefiles/ubsan.inc.mk

```diff
@@ -0,0 +1,61 @@
+# Copyright (C) 2019 Kaspar Schleiser <kaspar@schleiser.de>
+#
+# # Introduction
+#
+# This file contains makefile convenience support for gcc/clang's undefined
+# behaviour sanitizer.
+#
+# # Overview
+#
+# Both gcc and clang allow generation on code that does runtime checks for
+# undefined behavior (UB).
+#
+# E.g., the following code might trigger UB for some parameters:
+#
+#   void test(int foo) {
+#     return (foo << 24);
+#   }
+#
+#  In this case, the signed shift would be alright unless it would "push out"
+#  bits to the left, with undefined runtime result. Using ubsan, this can be
+#  caught.
+#
+# # How to use
+#
+# 1. build with `make all-ubsan`. This will cause a trap on undefined
+# behaviour.
+#
+# 2. build with `UBSAN_MODE=msg_exit make all-ubsan` will print where and how
+# the undefined behaviour occured, then exit with error code.  This is only
+# available on native built with gcc.
+#
```

Comment:
This won't get into Doxygen, unless I'm seriously ill-informed. Maybe this could be moved to a `doc.txt` so that this features becomes more visible?

Reply:
doc.txt syntax (markdown in C comments in .txt files) was too much. I've just added a new "debugging-aids.md" which gets included in the doxygen output.

Reply:
Thanks to an unholy Doxygen file type association, .mk files can have `##` lines with @defgroup / @ingroup.

Reply:
yeah, I found out about that while debugging the .md.
Anyhow, I'm not sure I prefer a .md or have it inline with the .mk. First is "with the docs", second is close to the code.
The makefile has a pointer to the .md, so in order to not have more work, I'd most prefer squashing and merging. ;)

Reply:
BTW, [here](https://15833-7370241-gh.circle-artifacts.com/0/doc/debugging-tools.html)'s the rendered docs from circleci.

Reply:
That was more a general pointer; docs [look good](https://15833-7370241-gh.circle-artifacts.com/0/doc/debugging-tools.html) as they are (see extra note, but from my limited PoV that's good for instasquash).